packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthErrorHandler.kt

```diff
@@ -76,6 +77,7 @@ class AuthErrorHandler {
                     is UnexpectedLambdaException -> errorCode = "LambdaException"
                     is UserLambdaValidationException -> errorCode = "LambdaException"
                     is TooManyFailedAttemptsException -> errorCode = "FailedAttemptsLimitExceededException"
+                    is LimitExceededException -> errorCode = "LimitExceededException"
```

Comment:
_Open Question_: Does this replaces the need for handling of `AuthException.LimitExceededException`

https://github.com/aws-amplify/amplify-flutter/blob/1ab9dc0b09b3e6dfa004d39beaae66f6688dd6d1/packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthErrorHandler.kt#L59

Reply:
After looking into this further, this is what I have found. 

The Android Amplify library can throw Cognito exceptions (exceptions from `com.amazonaws.services.cognitoidentityprovider`) and Amplify auth exceptions (exceptions from `com.amplifyframework.auth`).

The Android Amplify library will convert & map Cognito exceptions to specific Amplify auth exceptions for `signUp`, `confirmSignUp`, `resendSignUpCode`, `signIn`, and `confirmSignIn`. However, it will not convert & map the exception for any of the other methods (there are about a dozen other methods, for example: `fetchAuthSession`, `resetPassword`, `confirmResetPassword`, `updatePassword`, `fetchUserSessionAttributes`). It will instead throw a generic AuthException and nest the Cognito exception within the generic AuthException. This can be seen in [`AWSCognitoAuthPlugin.java`](https://github.com/aws-amplify/amplify-android/blob/main/aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java).

We are currently handling a subset of exceptions from both libraries. We are first checking to see if the exception is one of 18 Amplify auth exceptions, and mapping it to a specific error code if it is. If it is not any of those 18 Amplify auth exceptions, then we are checking to see if one of 9 Cognito exceptions is nested within it. That can be seen in the code above.

We currently only have a mapping for 18 of 20 Amplify Auth exceptions and 9 of 37 Cognito exceptions. `LimitExceededException` (added above) is one of the 28 Cognito exceptions that we don't have a mapping for.

To map all possible exceptions to specific exceptions, either the Android Amplify library would need to consistently throw an Amplify exception (and we would need to add mappings for the missing 2 exceptions here) or we need to map all 20 Amplify exceptions and all 37 Cognito exceptions.

Reply:
After discussing with @haverchuck, we decided to move forward with the fix above (pending a new unit test) and then add an item to the backlog to address this more holistically.